packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -153,4 +153,23 @@ export class CodeCommitSourceAction extends Action {
       },
     };
   }
+
+  private generateEventId(stage: codepipeline.IStage): string {
+    let branchIdDisambiguator: string;
+    let baseId = stage.pipeline.node.uniqueId;
+    if (this.branch === 'master') {
+      branchIdDisambiguator = baseId;
+    } else if (Token.isUnresolved(this.branch)) {
+      let candidate = baseId;
+      let counter = 0;
+      while (this.props.repository.node.tryFindChild(candidate) !== undefined) {
+        counter += 1;
+        candidate = `${baseId}-Branch${counter}-`;
+      }
+      branchIdDisambiguator = candidate;
+    } else {
+      branchIdDisambiguator = `${baseId}-${this.branch}-`;
+    }
+    return `${branchIdDisambiguator}EventRule`;
+  }
```

Comment:
I have a hard time keeping up with this code. For example, `branchIdDisambiguator` sounds to me like it should be just the branch-changing part of the ID, but somehow it's the entire thing.

What do you think of this instead?

```suggestion
  private generateEventId(stage: codepipeline.IStage): string {
    const baseId = stage.pipeline.node.uniqueId;
    if (Token.isUnresolved(this.branch)) {
      let candidate = '';
      let counter = 0;
      do {
        candidate = this.eventIdFromPrefix(`${baseId}-${counter}`);
        counter += 1;
      } while (this.props.repository.node.tryFindChild(candidate) !== undefined);
      return candidate;
    } else {
      const branchIdDisambiguator = this.branch === 'master' ? '' : '-${this.branch}-';
      return this.eventIdFromPrefix(`${baseId}${branchIdDisambiguator}`);
    }
  }
  
  private eventIdFromPrefix(eventIdPrefix: string) {
    return `${eventIdPrefix}EventRule`;
  }
```

(Notice the different loop; the current one actually checks the wrong ID, without the `EvenRule` suffix!)

Reply:
Oops, thanks for spotting that! I think that your naming scheme with the counter needs a slight tweak though, so I will just update my branch with that.

Reply:
@skinny85 This might include a bug:

```typescript
# Use buckticks instead
'-${this.branch}-' => `-${this.branch}-`
```